### PR TITLE
Remove names of unused parameters

### PR DESCRIPTION
**Pull request by tautschnig**

### CODE REVIEW

**Review by chrisr-diffblue — COMMENTED** [review]
While I can see the value to this change (I'm assuming it also helps reduce compiler warnings in VS), as far as I know it's not something that is part of the current coding standard... Is it therefore worth also updating CODING_STANDARD.md with this additional constraint? (I guess bonus points for updating linter/clang-format too :-) but I don't know if that is even supported by them)

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: 57002814250af52eb37da41d8c064066e2244d50).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/77253749
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).
##

**Review by kroening — APPROVED** [review]

**Comment by tautschnig** [comment]
@chrisr-diffblue I have created #2568 as a proposed amendment to the coding standard.

**Review by smowton — APPROVED** [review]

**Review by chrisr-diffblue — APPROVED** [review]
LGTM - thanks for the coding standards update too.

**Review by romainbrenguier — APPROVED** [review]

**Review by peterschrammel — APPROVED** [review]

**Comment by peterschrammel** [comment]
@tautschnig, sorry for the late review. Needs another rebase, then it is good to go.

File: src/goto-instrument/rw_set.h [hunk 1]
```diff
@@ -87,7 +87,7 @@ class rw_set_baset
   void output(std::ostream &out) const;
 
 protected:
-  virtual void track_deref(const entryt &entry, bool read) {}
+  virtual void track_deref(const entryt &, bool read) { (void)read; }
```
Comment (martin-cs) [inline src/goto-instrument/rw_set.h]: I'm probably missing some subtlety here but this one is not like it's friends...

**Review by martin-cs — APPROVED** [review]
+1 to what @chrisr-diffblue said.

Also, I fear this may need yet another rebase now that AIG has gone.

File: src/goto-symex/symex_catch.cpp [hunk 2]
```diff
@@ -11,8 +11,9 @@ Author: Daniel Kroening, kroening@kroening.com
 
 #include "goto_symex.h"
 
-void goto_symext::symex_catch(statet &state)
+void goto_symext::symex_catch(statet &)
 {
+  UNREACHABLE;
```
Comment (thk123) [inline src/goto-symex/symex_catch.cpp:16]: This should not have been put in the same commit as removing unused variables!! :angry:

Comment (tautschnig) [reply src/goto-symex/symex_catch.cpp:16]: Err, my apologies for failing to separate this one (and the one for `symex_fkt` now that I took another look) out. I suppose ... it's not actually `UNREACHABLE`?

Comment (thk123) [reply src/goto-symex/symex_catch.cpp:16]: Well it probably should be unreachable, but sadly we hit it. #2701 trips over this as a GOTO program in TG fails to do the `remove_exceptions` pass (but missing exceptions doesn't seem to cause any problems). I don't think this should block #2701 as it is somewhat incidental but I will fix and reinstate these.

Comment (tautschnig) [reply src/goto-symex/symex_catch.cpp:16]: I agree that it should not block #2701, but really this raises questions about code that still triggers this and the fact that it was able to survive for this long. My proposal is that, once #2701 is in, a new pull request be created that reverts the one commit from #2701. That new pull request should then also do further cleanup as those methods should go away completely.

**Pull request merged**